src/main/java/ua/vedroid/dao3/dao/jdbc/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,206 @@
+                + "WHERE cars.is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            List<Car> cars = new ArrayList<>();
+            while (resultSet.next()) {
+                cars.add(getCarFromResultSet(connection, resultSet));
+            }
+            return cars;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Get all cars is failed", ex);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET m_id = ?, model = ?"
+                + " WHERE id = ? AND is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            removeDrivers(connection, car.getId());
+            statement.setLong(1, car.getManufacturer().getId());
+            statement.setString(2, car.getModel());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+            if (!car.getDrivers().isEmpty()) {
+                addDrivers(connection, car);
+            }
+            return car;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Update " + car + " is failed", ex);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String query = "UPDATE cars SET is_deleted = true WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Delete of car with id="
+                    + id + " is failed", ex);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT * FROM cars c "
+                + "LEFT JOIN cars_drivers cd on c.id = cd.car_id "
+                + "LEFT JOIN drivers d on cd.driver_id = d.id "
+                + " LEFT JOIN manufacturers m on m.id = c.m_id "
+                + "WHERE d.id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            List<Car> cars = new ArrayList<>();
+            while (resultSet.next()) {
+                cars.add(getCarFromResultSet(connection, resultSet));
+            }
+            return cars;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("", ex);
```

Comment:
Everything is on fire and nobody knows why.

Reply:
agree